src/Sylius/Bundle/UiBundle/Twig/TestFormAttributeExtension.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\UiBundle\Twig;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class TestFormAttributeExtension extends AbstractExtension
+{
+    /** @var string */
+    private $env;
+
+    public function __construct(string $env)
+    {
+        $this->env = $env;
+    }
+
+    /**
+     * @return TwigFunction[]
+     */
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction(
+                'sylius_test_form_attribute',
+                function (string $name, ?string $value = null): array {
+                    if (strpos($this->env, 'test') === 0) {
+                        return ['attr' => ['data-test-'.$name => (string) $value]];
```

Comment:
```suggestion
                        return ['attr' => ['data-test-' . $name => (string) $value]];
```

Reply:
But in the case, we would like to introduce helper class, we can do
```suggestion
                        return ['attr' => TestAttribure::array($name, $value)];
```

Reply:
I'd say a helper class used only in one place without any possibility to replace is not a good idea, especially for maintainers.

Reply:
It won't be used only in one place. This class should be used in all behat scenarios. What I would like to achieve, is to have this string (and array) building logic in one class. 

Injecting it would be even better of course. The question is if it still is worth it when we will have to inject this service to every page. On the other hand, we can just extend parent class and add it as a protected field.

Reply:
Just to be clear. The same service (or helper class) should be used here and here: https://github.com/Sylius/Sylius/pull/10883/files#r352113248

Reply:
The output is different here:
 - for rendering we need:
   - HTML attribute: `data-test-key` or `data-test-key="value"`
   - form attribute: `['attr' => ['data-test-key' => '']]` or `['attr' => ['data-test-key' => 'value']]`
 - for matching we need:
   - either purely key-based: `[data-test-key]`
   - or basing on key and value: `[data-test-key="value"]`

If we go with static class, I'd only implement two methods:
 - one that creates the identifier, passing `key` as a parameter would return string `data-test-key`
 - one that creates the matcher, passing `key` as a parameter would return string  `[data-test-key]` and passing `key` and `value` would return string `[data-test-key="value"]`

I don't see where method returning array `['data-test-'.$key => (string) $value]` for `$key, $value` could be reused.

Reply:
I thought about putting all creation methods into one class, but anything will be better than writing strings